projects/driver_controls/src/mech_brake_calibration.c

```diff
@@ -0,0 +1,84 @@
+#include <limits.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <stdbool.h>
+#include "ads1015.h"
+#include "delay.h"
+#include "gpio_it.h"
+#include "i2c.h"
+#include "interrupt.h"
+#include "log.h"
+#include "mech_brake_calibration.h"
+#include "soft_timer.h"
+#include "status.h"
+#include "unity.h"
+#include "wait.h"
+
+static void prv_callback_channel(Ads1015Channel ads1015_channel, void *context) {
+  MechBrakeCalibrationStorage *storage = context;
+
+  MechBrakeCalibrationPointData *data = &storage->data[storage->sample_point];
+  if (data->sample_counter >= MECH_BRAKE_CALIBRATION_NUM_SAMPLES) {
+    ads1015_configure_channel(storage->settings.ads1015, ads1015_channel, false, NULL, NULL);
+  } else {
+    int16_t reading = 0;
+    ads1015_read_raw(storage->settings.ads1015, ads1015_channel, &reading);
+
+    data->min_reading = MIN(data->min_reading, reading);
+    data->max_reading = MAX(data->max_reading, reading);
+  }
+}
+
+StatusCode mech_brake_calibration_init(MechBrakeCalibrationStorage *storage,
+                                       MechBrakeSettings *settings) {
+  memset(storage, 0, sizeof(*storage));
+  storage->settings = *settings;
+
+  return STATUS_CODE_OK;
+}
+
+StatusCode mech_brake_sample(MechBrakeCalibrationStorage *storage,
+                             MechBrakeCalibrationPoint point) {
+  // Disables channel
+  ads1015_configure_channel(storage->settings.ads1015, storage->settings.channel, false, NULL,
+                            NULL);
+
+  storage->data[point] = (MechBrakeCalibrationPointData){
+    .sample_counter = 0,
+    .min_reading = INT16_MAX,
+    .max_reading = INT16_MIN,
+  };
+  storage->sample_point = point;
+
+  // enables channel
+  ads1015_configure_channel(storage->settings.ads1015, storage->settings.channel, true,
+                            prv_callback_channel, storage);
+
+  bool samples_completed = false;
+  do {
+    wait();
+    samples_completed = true;
+    samples_completed &=
```

Comment:
Please use your own code instead of copy and pasting from the throttle. This is unnecessarily complex.